rct229/rulesets/ashrae9012019/section19/section19rule14.py

```diff
@@ -0,0 +1,273 @@
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1.2.8.1 Excluding Exceptions 1 and 2",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+
+        dict_of_zones_and_terminal_units_served_by_hvac_sys_b = (
+            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmi_b)
+        )
+        zone_supply_return_exhaust_relief_terminal_fan_power_dict = (
+            get_zone_supply_return_exhaust_relief_terminal_fan_power_dict(rmi_p)
+        )
+
+        hvac_info = {}
+        for hvac_b in find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+            rmi_b,
+        ):
+            hvac_id_b = hvac_b["id"]
+            hvac_info[hvac_id_b] = {}
+
+            fan_system_info_b = (
+                get_zone_supply_return_exhaust_relief_terminal_fan_power_dict(
+                    getattr_(hvac_b, "HVAC", "fan_system")
+                )
+            )
+
+            hvac_info[hvac_id_b]["more_than_one_supply_or_return_fan"] = (
+                False
+                if (
+                    fan_system_info_b["supply_fans_qty"] == 1
+                    and fan_system_info_b["return_fans_qty"] == 1
+                )
+                else True
+            )
+
+            for zone_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b[
+                hvac_id_b
+            ]["zone_list"]:
+                modeled_fan_power_list_p = list(
+                    zone_supply_return_exhaust_relief_terminal_fan_power_dict[
+                        find_one(
+                            f'$.buildings[*].building_segments[*].zones[?(@.id == "{zone_b}")]',
+                            rmi_p,
+                        )["id"]
+                    ]
+                )
+
+                hvac_info[hvac_id_b]["is_modeled_with_return_fan_in_proposed"] = (
+                    True
+                    if (
+                        modeled_fan_power_list_p["zone_total_return_fan_power"]
+                        > ZERO.POWER
+                    )
+                    else False
+                )
+
+                hvac_info[hvac_id_b]["is_modeled_with_relief_fan_in_proposed"] = (
+                    False
+                    if (
+                        modeled_fan_power_list_p["zone_total_exhaust_fan_power"]
+                        > ZERO.POWER
+                    )
+                    else True
+                )
```

Comment:
I think you can pass this down so in here what you need to do is:
```python
zone_fan_power_dict = {}
for zone_id_b in find_all("$.buildings[*].building_segments[*].zones[*].id, rmi_b)
    modeled_fan_power_list_p = list(
                    zone_supply_return_exhaust_relief_terminal_fan_power_dict[
                        find_one(
                            f'$.buildings[*].building_segments[*].zones[?(@.id == "{zone_id_b}")]',
                            rmi_p,
                        )["id"]
                    ]
                )
    zone_fan_power_dict [zone_id] = modeled_fan_power_list_p 
....
return {
    "baseline_system_type_dict": get_baseline_system_types(rmi_b),
    "dict_of_zones_and_terminal_units_served_by_hvac_sys_b": dict_of_zones_and_terminal_units_served_by_hvac_sys_b,
    "zone_fan_power_dict": zone_fan_power_dict , 
}
```

Reply:
Addressed.